README.md

```diff
@@ -56,6 +56,39 @@ public class MyController : Controller
 }
 ```
 
+You can also pass special options that apply when the proxy operation occurs.
+
+```csharp
+public class MyController : Controller
+{
+    [Route("api/posts/{postId}")]
+    public Task GetPosts(int postId)
+    {
+        var options = ProxyOptions.Instance
+            .WithShouldAddForwardedHeaders(false)
+            .WithBeforeSend((c, hrm) =>
+            {
+                // Set something that is needed for the downstream endpoint.
+                hrm.Headers.Authorization = new AuthenticationHeaderValue("Basic");
+            })
+            .WithAfterReceive((c, hrm) =>

```

Comment:
I understad what's wrong with this: you can't _partially change_ Content, you can just fully replace it. But for partially change you shoulr read Content, and for do this you should use async methods, so Action seems a bit not correct type for callback, it's more like `Func<context, response, Task>`

Reply:
Yeah, I have been going back and forth on this.  I think the scenario you mention is compelling, so I will change this.